packages/scapi/src/index.ts

```diff
@@ -133,8 +133,8 @@ export default class SCAPI extends events.EventEmitter {
   }
 
   receive(signal, msg): void {
```

Comment:
This would be how the types could have prevented the error:

```ts
import { packMessage, unpackMessage, OSCMessage } from "@supercollider/osc";

//...
receive(signal: string, msg: OSCMessage): void {

}
```

Then it would have thrown a compile error with `msg.args[1].value`

I didn't do all the strict typing yet. I will tighten it up each time I work on any of the packages.

Reply:
https://github.com/crucialfelix/supercolliderjs/blob/develop/packages/osc/src/types.ts#L51

Reply:
Just checked if I might be able to add this to the PR - unfortunately it is not quite this easy because the `requestId` is being used as the index to `this.requests` and since it is a general `OSCMessage` type, the `requestId` may be one of various types.  See compile error below.

Given this is the case, would you consider merging this PR as-is to eliminate the critical error with the API Quark?

If you'd propose the outline of a fix I'm happy to enhance my typescript fu to implement some of this stuff.


```
$ npm run build

> supercolliderjs@1.0.0-alpha.0 build /Users/colin/Projects/patches/supercolliderjs
> lerna exec -- tsc -b .

lerna notice cli v3.18.1
lerna info versioning independent
lerna info Executing command in 8 packages: "tsc -b ."
src/index.ts:138:35 - error TS2538: Type 'Buffer' cannot be used as an index type.

138     const request = this.requests[requestId];
                                      ~~~~~~~~~

src/index.ts:138:35 - error TS2538: Type '[string, ...CompletionMsgItem[]]' cannot be used as an index type.

138     const request = this.requests[requestId];
                                      ~~~~~~~~~

src/index.ts:138:35 - error TS2538: Type 'false' cannot be used as an index type.

138     const request = this.requests[requestId];
                                      ~~~~~~~~~

src/index.ts:138:35 - error TS2538: Type 'null' cannot be used as an index type.

138     const request = this.requests[requestId];
                                      ~~~~~~~~~

src/index.ts:138:35 - error TS2538: Type 'true' cannot be used as an index type.

138     const request = this.requests[requestId];
                                      ~~~~~~~~~

src/index.ts:148:29 - error TS2345: Argument of type 'OscType' is not assignable to parameter of type 'string'.
  Type 'null' is not assignable to type 'string'.

148         result = JSON.parse(result);
                                ~~~~~~

src/index.ts:149:18 - error TS2531: Object is possibly 'null'.

149         result = result.result;
                     ~~~~~~

src/index.ts:149:25 - error TS2339: Property 'result' does not exist on type 'string | number | boolean | Buffer | [string, ...CompletionMsgItem[]]'.
  Property 'result' does not exist on type 'string'.

149         result = result.result;
                            ~~~~~~

src/index.ts:168:26 - error TS2538: Type 'Buffer' cannot be used as an index type.

168     delete this.requests[requestId];
                             ~~~~~~~~~

src/index.ts:168:26 - error TS2538: Type '[string, ...CompletionMsgItem[]]' cannot be used as an index type.

168     delete this.requests[requestId];
                             ~~~~~~~~~

src/index.ts:168:26 - error TS2538: Type 'false' cannot be used as an index type.

168     delete this.requests[requestId];
                             ~~~~~~~~~

src/index.ts:168:26 - error TS2538: Type 'null' cannot be used as an index type.

168     delete this.requests[requestId];
                             ~~~~~~~~~

src/index.ts:168:26 - error TS2538: Type 'true' cannot be used as an index type.

168     delete this.requests[requestId];
                             ~~~~~~~~~


Found 13 errors.
```

Reply:
Certainly. I figured you would enjoy fiddling with ts, but let's merge and move forward. I or you can revisit the types later.

Reply:
My docs branch is close to merging. I would like to do the next alpha including the new docs. Today or tomorrow.